### PR TITLE
Allow configuring credentials for custom Gradle plugin repository

### PR DESCRIPTION
**Pull request by welandaz**

### CODE REVIEW

File: sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle [hunk 1]
```diff
@@ -23,6 +23,8 @@ initscript {
     def pluginRepositoryUrl = getInputParam('gradle.plugin-repository.url')
     def gePluginVersion = getInputParam('develocity.plugin.version')
     def ccudPluginVersion = getInputParam('develocity.ccud-plugin.version')
+    def gradlePluginRepoUsername = getInputParam('gradle.plugin-repository.username')
+    def gradlePluginRepoPassword = getInputParam('gradle.plugin-repository.password')
```
Comment (erichaagdev) [inline sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle:27]: I would rename the variables as `pluginRepositoryUsername` and `pluginRepositoryPassword` to be consistent with the existing `pluginRepositoryUrl`. 

Then I would reorder the variable definitions so all three are next to each other.

**Review by erichaagdev — CHANGES_REQUESTED** [review]

File: setup-gradle/README.md [hunk 2]
```diff
@@ -663,15 +663,17 @@ with a valid [Develocity access key](https://docs.gradle.com/enterprise/gradle-p
...
+| DEVELOCITY_PLUGIN_VERSION         | :white_check_mark: | the version of the [Develocity Gradle plugin](https://docs.gradle.com/enterprise/gradle-plugin/) to apply                                                                |
+| DEVELOCITY_CCUD_PLUGIN_VERSION    |  | the version of the [Common Custom User Data Gradle plugin](https://github.com/gradle/common-custom-user-data-gradle-plugin) to apply, if any                             |
+| GRADLE_PLUGIN_REPOSITORY_URL      |  | the URL of the repository to use when resolving the Develocity and CCUD plugins; the Gradle Plugin Portal is used by default                                             |
+| GRADLE_PLUGIN_REPOSITORY_USERNAME |  | the username for the repository URL to use when resolving the Develocity and CCUD plugins                                                                                |
+| GRADLE_PLUGIN_REPOSITORY_PASSWORD |  | the password for the repository URL to use when resolving the Develocity and CCUD plugins                                                                                |
```
[the remark this answers is not in the document]
Comment (welandaz) [reply setup-gradle/README.md]: Added

File: sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle (hunk 1 above)
Comment (welandaz) [reply sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle]: Done

File: sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle [hunk 3]
```diff
@@ -32,7 +34,19 @@ initscript {
         logger.quiet("Develocity plugins resolution: $pluginRepositoryUrl")
 
         repositories {
-            maven { url pluginRepositoryUrl }
+            maven {
+                url pluginRepositoryUrl
+                if (pluginRepoUsername && pluginRepoPassword) {
+                    logger.quiet("Using credentials for plugin repository")
```
Comment (erichaagdev) [inline sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle:40]: With f091a59e673b494634fcadd853e89a57a363351e merged I think this should be logged at `lifecycle`.

File: sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle [hunk 4]
```diff
@@ -21,6 +21,8 @@ initscript {
     }
 
     def pluginRepositoryUrl = getInputParam('gradle.plugin-repository.url')
+    def pluginRepoUsername = getInputParam('gradle.plugin-repository.username')
+    def pluginRepoPassword = getInputParam('gradle.plugin-repository.password')
```
Comment (erichaagdev) [inline sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle:25]: I would spell out `repository` so it is the same as the existing `pluginRepositoryUrl`.

```suggestion
    def pluginRepositoryUsername = getInputParam('gradle.plugin-repository.username')
    def pluginRepositoryPassword = getInputParam('gradle.plugin-repository.password')
```

**Review by erichaagdev — APPROVED** [review]

**Pull request merged**